jupyter_elfinder/elfinder.py

```diff
@@ -391,28 +360,112 @@ def run(
 
         return self.http_status_code, self.http_header, self.http_response
 
+    def __places(self) -> None:
+        if "targets[]" not in self._request:
+            self._response["error"] = "Invalid parameters"
+            return
+
+        targets = self._request["targets[]"]
+        files = []
+        for target in targets:
+            path = self.__find(target)
+            if path is None:
+                self.__set_error_data(target, "File not found")
+            else:
+                files.append(self.__info(path))
+        self._response["files"] = files
+
     def __open(self) -> None:
         """Open file or directory."""
-        # try to open file
-        if "tree" not in self._request and "current" in self._request:
-            cur_dir = self.__find_dir(self._request["current"], None)
+        # try dir
 
-            if not cur_dir:
-                self.http_status_code = 404
-                self.http_header["Content-type"] = "text/html"
-                self.http_response = "File not found"
+        # initialized = len(self._cached_path) > 0
+        if "init" in self._request and self._request["init"]:
+            self._response["api"] = 2.1
+        else:
+            if "target" not in self._request:
+                self._response["error"] = "Invalid parameters"
                 return
 
-            cur_file = self.__find(self._request["target"], cur_dir)
+        if "target" in self._request and self._request["target"]:
+            target = self.__find_dir(self._request["target"])
+            if not target:
+                self._response["error"] = (
+                    "Invalid parameters: " + self._request["target"]
+                )
+            elif not self.__is_allowed(target, "read"):
+                self._response["error"] = "Access denied"
```

Comment:
If we get to here, don't we want to return after this? Should we really continue on error?

Reply:
You are right, we should return.